packages/graph-explorer/src/core/ConfigurationProvider/useConfiguration.ts

```diff
@@ -1,137 +1,125 @@
 import { selector, selectorFamily, useRecoilValue } from "recoil";
-import DEFAULT_ICON_URL from "@/utils/defaultIconUrl";
-import { mergedConfigurationSelector } from "@/core/StateProvider/configuration";
-import type {
-  ConfigurationContextProps,
-  EdgeTypeConfig,
-  VertexTypeConfig,
-} from "./types";
+import {
+  allEdgeTypeConfigsSelector,
+  allVertexTypeConfigsSelector,
+  getDefaultEdgeTypeConfig,
+  getDefaultVertexTypeConfig,
+  mergedConfigurationSelector,
+} from "@/core/StateProvider/configuration";
+import type { ConfigurationContextProps } from "./types";
 
-function getDefaultVertexTypeConfig(vertexType: string): VertexTypeConfig {
-  return {
-    color: "#128EE5",
-    iconUrl: DEFAULT_ICON_URL,
-    iconImageType: "image/svg+xml",
-    type: vertexType,
-    displayLabel: "",
-    attributes: [],
-  };
-}
-
-function getDefaultEdgeTypeConfig(edgeType: string): EdgeTypeConfig {
-  return {
-    type: edgeType,
-    displayLabel: "",
-    attributes: [],
-  };
-}
+const assembledConfigSelector = selector<ConfigurationContextProps | undefined>(
+  {
+    key: "assembled-config",
+    get: ({ get }) => {
+      const configuration = get(mergedConfigurationSelector);
+      if (!configuration) {
+        return;
+      }
 
-export const assembledConfigSelector = selector<
-  ConfigurationContextProps | undefined
->({
-  key: "assembled-config",
-  get: ({ get }) => {
-    const configuration = get(mergedConfigurationSelector);
-    if (!configuration) {
-      return;
-    }
+      const vertexTypesMap = new Map(
```

Comment:
Nit: It seems this map is only used to obtain unique vertex types via the keys of the map - would it be sufficient to use a Set instead of a Map since the map values are not used?

Reply:
At this point, yes. I missed this during my refactoring to remove the functions. Thanks! I'll update.